packages/pluggableWidgets/color-picker-web/src/components/ColorPicker.tsx

```diff
@@ -0,0 +1,152 @@
+import { getColorPicker, parseColor, validateColorFormat } from "../utils";
+import { Input } from "./Input";
+import { Button } from "./Button";
+
+export interface ColorPickerProps {
+    id: string;
+    name: string;
+    tabIndex?: number | undefined;
+    onChange: () => void;
+    onChangeComplete: (value: string) => void;
+    mode: ModeEnum;
+    type: TypeEnum;
+    color: string;
+    defaultColors: DefaultColorsType[];
+    format: FormatEnum;
+    disabled: boolean;
+    invalidFormatMessage?: string;
+}
+
+export type ColorPickerConfigProps =
+    | SketchPickerProps
+    | ChromePickerProps
+    | TwitterPickerProps
+    | HuePickerProps
+    | BlockPickerProps
+    | GithubPickerProps
+    | CirclePickerProps
+    | SwatchesPickerProps
+    | SliderPickerProps
+    | MaterialPickerProps
+    | CompactPickerProps;
+export const ColorPicker = (props: ColorPickerProps): ReactElement => {
+    const defaultColor = {
+        hex: "#FFFFFF",
+        rgb: "rgb(255,255,255)",
+        rgba: "rgb(255,255,255,1)"
+    };
+    const { type, mode, disabled, defaultColors, color, format, invalidFormatMessage } = props;
+    const colorElement = getColorPicker(type);
+    const [hidden, setHidden] = useState(mode !== "inline");
+    const [config, setConfig] = useState<ColorPickerConfigProps>({});
+    const [currentColor, setCurrentColor] = useState<string>(color || defaultColor[format]);
+    const [alertMessage, setAlertMessage] = useState<string | undefined>();
+
+    const validateColor = (color: string): void => {
+        const validFormat = validateColorFormat(color, format);
+        const alertMessage = validFormat ? invalidFormatMessage?.replace(/\{1}/, validFormat) : undefined;
+        setAlertMessage(alertMessage);
+    };
+    const onToggle = useCallback(
+        (hide: boolean): void => {
+            if (!disabled && mode !== "inline") {
+                setHidden(hide);
+            }
+        },
+        [disabled, mode]
+    );
+    const onColorValueUpdate = useCallback(
+        (color: ColorState): void => {
+            if (color && !disabled) {
+                const finalColor = parseColor(color, format);
+                setCurrentColor(finalColor);
+                props.onChangeComplete(finalColor);
+            }
+        },
+        [disabled, format, props]
+    );
+
+    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        const value = event.target.value;
+        validateColor(color);
+        setCurrentColor(value);
+        props.onChangeComplete(value);
+    };
+
+    const renderInput = (): ReactElement => {
+        return (
+            <Input color={color} disabled={disabled} onKeyUp={onToggle} onChange={onInputChange}>
+                {renderButton()}
+            </Input>
+        );
+    };
+    const renderButton = (): ReactElement => {
+        return <Button mode={mode} hidden={hidden} disabled={disabled} onClick={onToggle} color={color} />;
```

Comment:
Continuing my `Button` comment: you can do `onClick={() => onToggle(!hidden)}` here and it will by the same as what you have right now (as far as I can see) + you can get rid of one API.